// Copyright 2023 Ant Group Co., Ltd.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//   http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//

syntax = "proto3";

package secretflow.serving;

message OneHotSubstitutionSpec {
  message FeatureMapping {
    message Category {
      string mapping_name = 1;
      string value = 2;
    }
    string raw_name = 1;
    repeated Category categories = 2;
  }
  repeated FeatureMapping feature_mappings = 1;
}
